middle_end/flambda/simplify/simplify_named.rec.ml

```diff
@@ -26,8 +26,8 @@ type simplify_named_result =
   | Reified of {
       definition : Named.t;
       bound_symbol : Bound_symbols.t;
+      symbol : Symbol.t;
```

Comment:
This is redundant with `bound_symbol` above. Would it make sense for `DA.consider_constant_for_sharing` to take a `Bound_symbols.t` instead of a `Symbol.t`, so that you can reuse `bound_symbol` directly ?

Reply:
This will be fixed once #224 is merged, which deletes `bound_symbol` from this record.